beacon-chain/core/state/state_test.go

```diff
@@ -11,6 +12,169 @@ import (
+		t.Error("ValidatorBalances was not correctly initialized")
+	}
+
+	// Randomness and committees fields checks.
+	if len(state.LatestRandaoMixesHash32S) != latestRandaoMixesLength {
+		t.Error("Length of LatestRandaoMixesHash32S was not correctly initialized")
+	}
+	if len(state.LatestVdfOutputs) != LatestVdfMixesLength {
+		t.Error("Length of LatestRandaoMixesHash32S was not correctly initialized")
+	}
+	if len(state.PersistentCommittees) != shardCount {
+		t.Error("PersistentCommittees was not correctly initialized")
+	}
+	if !reflect.DeepEqual(state.PersistentCommitteeReassignments, []*pb.ShardReassignmentRecord{}) {
+		t.Error("PersistentCommitteeReassignments was not correctly initialized")
+	}
+
+	// Proof of custody field check.
+	if !reflect.DeepEqual(state.PocChallenges, []*pb.ProofOfCustodyChallenge{}) {
+		t.Error("PocChallenges was not correctly initialized")
+	}
+
+	// Finality fields checks.
+	if state.PreviousJustifiedSlot != initialSlotNumber {
+		t.Error("PreviousJustifiedSlot was not correctly initialized")
+	}
+	if state.JustifiedSlot != initialSlotNumber {
+		t.Error("JustifiedSlot was not correctly initialized")
+	}
+	if state.FinalizedSlot != initialSlotNumber {
+		t.Error("FinalizedSlot was not correctly initialized")
+	}
+	if state.JustificationBitfield != 0 {
+		t.Error("JustificationBitfield was not correctly initialized")
+	}
+
+	// Recent state checks.
+	if len(state.LatestCrosslinks) != shardCount {
+		t.Error("Length of LatestCrosslinks was not correctly initialized")
+	}
+	if len(state.LatestBlockRootHash32S) != latestBlockRootsLength {
+		t.Error("Length of LatestBlockRootHash32S was not correctly initialized")
+	}
+	if !reflect.DeepEqual(state.LatestPenalizedExitBalances, []uint64{}) {
+		t.Error("LatestPenalizedExitBalances was not correctly initialized")
+	}
+	if !reflect.DeepEqual(state.LatestAttestations, []*pb.PendingAttestationRecord{}) {
+		t.Error("LatestAttestations was not correctly initialized")
+	}
+	if !reflect.DeepEqual(state.BatchedBlockRootHash32S, [][]byte{}) {
+		t.Error("BatchedBlockRootHash32S was not correctly initialized")
+	}
+
+	// PoW receipt root checks.
+	if !bytes.Equal(state.ProcessedPowReceiptRootHash32, processedPowReceiptRoot) {
+		t.Error("ProcessedPowReceiptRootHash32 was not correctly initialized")
+	}
+	if !reflect.DeepEqual(state.CandidatePowReceiptRoots, []*pb.CandidatePoWReceiptRootRecord{}) {
+		t.Error("CandidatePowReceiptRoots was not correctly initialized")
+	}
+
+	// Initial committee shuffling check.
+	if len(state.ShardAndCommitteesAtSlots) != int(2*epochLength) {
+		t.Error("ShardAndCommitteesAtSlots was not correctly initialized")
+	}
+
+	for i := 0; i < len(state.ShardAndCommitteesAtSlots); i++ {
+		if len(state.ShardAndCommitteesAtSlots[i].ArrayShardAndCommittee[0].Committee) !=
+			int(params.BeaconConfig().TargetCommitteeSize) {
+			t.Error("ShardAndCommittees was not correctly initialized")
+		}
+	}
+}
+
 func TestGenesisState_HashEquality(t *testing.T) {
```

Comment:
With this InitialGenesisState function, you should also remove  NewGenesisState and replace all references to it with the new function, since these two functions fill the same purpose

Reply:
Will do that in the next PR. Don't want the diff to get too big.